src/native/libs/System.Globalization.Native/pal_locale.c

```diff
@@ -134,10 +134,17 @@ int32_t FixupLocaleName(UChar* value, int32_t valueLength)
 // is not desirable at all because it doesn't support case insensitive string comparisons.
 const char* DetectDefaultLocaleName()
 {
-    const char* icuLocale = uloc_getDefault();
+    const char* icuLocale;
+
+    icuLocale = uloc_getDefault();
+
     if (strcmp(icuLocale, "en_US_POSIX") == 0)
     {
+#ifdef TARGET_OS_MAC
```

Comment:
should be `__APPLE__` to match what we use in the header
```suggestion
#ifdef __APPLE__
```

Reply:
please also update the comment at the top of this function to explain the different behavior on Apple platforms